packages/jaspr/lib/src/browser/component_anchors.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:universal_web/web.dart' as web;
+
+import '../../jaspr.dart';
+import 'options.dart';
+
+sealed class ComponentAnchor {
+  ComponentAnchor(this.name, this.startNode);
+
+  final String name;
+  final web.Node startNode;
+  late web.Node endNode;
+}
+
+class ClientComponentAnchor extends ComponentAnchor {
+  ClientComponentAnchor(super.name, super.startNode, this.data);
+
+  String? data;
+  late FutureOr<ClientBuilder> builder;
+
+  late final params = data != null
+      ? jsonDecode(const DomValidator().unescapeMarkerText(data!)) as Map<String, dynamic>
+      : <String, dynamic>{};
+
+  Future<void> resolve() async {
+    var r = await builder;
+    builder = r;
+  }
+
+  Component build() {
+    assert(builder is ClientBuilder, "ClientComponentAnchor was not resolved before calling build()");
+    return (builder as ClientBuilder)(params);
+  }
+}
+
+final _clientStartRegex = RegExp('^${DomValidator.clientMarkerPrefixRegex}(\\S+)(?:\\s+data=(.*))?\$');
+final _clientEndRegex = RegExp('^/${DomValidator.clientMarkerPrefixRegex}(\\S+)\$');
```

Comment:
Unnecessary as part of this change, but should consider moving these away from regex.